packages/ra-core/src/controller/useReference.ts

```diff
@@ -1,12 +1,14 @@
 import { RaRecord, Identifier } from '../types';
 import { UseGetManyHookValue, useGetManyAggregate } from '../dataProvider';
+import { UseQueryOptions } from 'react-query';
 
-interface UseReferenceProps {
+interface UseReferenceProps<RecordType extends RaRecord = RaRecord> {
     id: Identifier;
     reference: string;
+    options?: UseQueryOptions<RecordType[], Error>;
 }
 
-export interface UseReferenceResult<RecordType extends RaRecord = any> {
+export interface UseReferenceResult<RecordType extends RaRecord = RaRecord> {
```

Comment:
why did you change that? The signature of all our other hooks uses `any` as default value.

Reply:
we did the opposite when refactoring for 4.0 everywhere... this component shouldn't take another direction on its own.

Reply:
cf https://github.com/marmelab/react-admin/pull/7078